server/controllers/shoutoutController.js

```diff
@@ -31,14 +29,24 @@ const shoutoutController = {
   getAllShoutOuts: async () => {
     return await db('shoutouts')
       .select()
-      .then((shoutouts) => {
-        const shoutoutPromises = shoutouts
-          .sort((a, b) => b.date - a.date)
-          .map((shout) => createShoutoutRes(db, shout));
-        return Promise.all(shoutoutPromises);
-      })
+      .orderBy('date', 'desc')
+      .then((shouts) =>
+        Promise.all(shouts.map((shout) => createShoutoutRes(db, shout)))
+      )
       .then((formattedShoutouts) => formattedShoutouts);
   },
+  getShoutoutsByuserId: async (type, userId, days) => {
+    const lastDay = new Date();
+    const firstDay = new Date(new Date().setDate(lastDay.getDate() - days));
+
+    return await db('shoutouts')
+      .whereBetween('date', [firstDay, lastDay])
+      .where(type, userId)
+      .orderBy('date', 'desc')
```

Comment:
I think the Month picker can be used here for the FE to pass the firstDay and last day of the two selected months. Something to think about as we build out that functionality for the User's table.